x/models.py

```diff
@@ -176,3 +180,73 @@ def input_shape(self):
     def description(self):
         dstr = "Keras Model \n\t Optimizer: {} \n\t Loss: {} \n\t Policy: {}"
         return dstr.format(self.optimizer, self.loss, self.policy_rule)
+
+
+class TableModel(Model):
+    def __init__(self, state_dim, num_actions):
+        """Table model
+        Stores and updates a Q-value for every possible state/action pair
+        Attributes
+        ----------
+        state_dim : dimension of state space
+        num_actions: size of action space
+
+        """
+        self.state_dim = state_dim
+        self.n_actions = num_actions
+
+    def compile(self, state_dim_values, lr=0.2, policy_rule="maxrand", init_value=None):
```

Comment:
This proves that we need our own `optimizers.py` file. In Keras, when the "sgd" string comes in, there is a rule somewhere that says where to look for the rules and everything. For this model, "sgd" is hard coded. There is no way to use another algorithm (say RMSprop) in here.

But we should do this later, for now I don't have anything else to say against this model. Perhaps the fact that different models will have different APIs. Do you think there is a way to unify them?